rust/xaynet-core/src/mask/object/serialization.rs

```diff
@@ -202,6 +210,38 @@ impl FromBytes for MaskObject {
 
         Ok(MaskObject { data, config })
     }
+
+    fn from_byte_stream<I: Iterator<Item = u8> + ExactSizeIterator>(
+        iter: &mut I,
+    ) -> Result<Self, DecodeError> {
+        let config = MaskConfig::from_byte_stream(iter)?;
+        if iter.len() < 4 {
```

Comment:
I'm not sure that this line will ever be executed because by calling `MaskConfig::from_byte_stream(iter)?`
`MaskConfigBuffer::check_buffer_length` would already return the error `invalid buffer length`.

Reply:
No, because `iter` is mutated. `MaskConfig::from_byte_stream(iter)?` consumes a few bytes, and then we check that there are at least 4 bytes left in `iter`.

Reply:
Ah you are right👍 my bad